### PR TITLE
[feature](merge-cloud) Add multi cloud cluster

### PR DESCRIPTION
**Pull request by yujun777**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 1]
```diff
@@ -1876,6 +1937,9 @@ protected void readFields(DataInput in) throws IOException {
             }
         }
         if (Env.getCurrentEnvJournalVersion() >= FeMetaVersion.VERSION_117) {
+            if (Config.isCloudMode()) {
+                cloudClusterId = Text.readString(in);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java:1941]: @w41ter  pls take a look at this change

Comment (w41ter) [reply fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java:1941]: It seems that this field was added after `VERSION_123` in selectdb. Please add another condition to check whether to read this field.

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 2]
```diff
@@ -708,6 +720,10 @@ private void handleQueryWithRetry(TUniqueId queryId) throws Exception {
         syncJournalIfNeeded();
 
         int retryTime = Config.max_query_retry_time;
+        if (Config.isCloudMode()) {
+            // be core and be restarted, need retry more times
+            retryTime = Config.cloud_query_failed_retry_times;
```
Comment (deardeng) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:725]: 可以继续复用max_query_retry_time。当时不想和社区的搅合一起。所以新开个新的参数。现在要往里面合的话，可以原来的那个配置

File: fe/fe-core/src/main/java/org/apache/doris/system/Backend.java [hunk 3]
```diff
@@ -277,6 +277,22 @@ public void setLoadDisabled(boolean isLoadDisabled) {
...
+
+    public boolean isActive() {
+        return this.backendStatus.isActive;
+    }
+
+    public long getCurrentFragmentNum() {
+        return this.backendStatus.currentFragmentNum;
+    }
+
+    public long getLasetCurrentFragmentNum() {
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/system/Backend.java:292]: A better name should be provided.

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 4]
```diff
@@ -718,14 +730,62 @@ private void handleQueryWithRetry(TUniqueId queryId) throws Exception {
...
                     context.setQueryId(newQueryId);
+                    if (Config.isCloudMode()) {
+                        // sleep random millis [1000, 1500] ms
+                        // in the begining of retryTime/2
+                        int randomMillis = 1000 + (int) (Math.random() * (1000 - 500));
+                        LOG.debug("stmt executor retry times {}, wait randomMillis:{}, stmt:{}",
+                                i, randomMillis, originStmt.originStmt);
+                        try {
+                            if (i > retryTime / 2) {
+                                // sleep random millis [2000, 2500] ms
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:741]: We should handle same with cloud mode diabled.

**Review by dataroaring — COMMENTED** [review]
some comments.

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java (hunk 1 above)
Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java]: add check version >= 123

File: fe/fe-core/src/main/java/org/apache/doris/system/Backend.java (hunk 3 above)
Comment (yujun777) [reply fe/fe-core/src/main/java/org/apache/doris/system/Backend.java:292]: this function no use. deleted

**Comment by dataroaring** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run cloud_p0

**Comment by yujun777** [comment]
run cloud_p1

**Comment by yujun777** [comment]
run p0

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by yujun777** [comment]
run p0

**Comment by yujun777** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Pull request merged**